tests/NationalInstruments.Analyzers.UnitTests/LongChainDottedInvocationAnalyzerTests.cs

```diff
@@ -0,0 +1,331 @@
+                .GroupBy(software => software.AliasName)
+                .Select(g => g.First())
+                .Distinct();
+        }
+
+        private interface ISoftwareContent
+        {
+            string AliasName { get; }
+        }
+    }
+}
+"
+                }, // inside a method
+                new object[]
+                {
+                    @"
+using System.Collections.Generic;
+using System.Linq;
+
+namespace NationalInstruments.Analyzers.UnitTests
+{
+    class Test
+    {
+        Test(IEnumerable<ISoftwareContent> selectedSoftwareContents)
+        {
+            var distinctSelectedSoftware = selectedSoftwareContents
+                .GroupBy(software => software.AliasName)
+                .Select(g => g.First())
+                .Distinct();
+        }
+
+        private interface ISoftwareContent
+        {
+            string AliasName { get; }
+        }
+    }
+}
+"
+                }, // inside a constructor
+                new object[]
+                {
+                    @"
+using System.Collections.Generic;
+using System.Linq;
+
+namespace NationalInstruments.Analyzers.UnitTests
+{
+    class Test
+    {
+        private IEnumerable<ISoftwareContent> selectedSoftwareContents = Enumerable.Empty<ISoftwareContent>();
+
+        IEnumerable<ISoftwareContent> DistinctSoftwareContents =>
+         selectedSoftwareContents
+            .GroupBy(software => software.AliasName)
+            .Select(g => g.First())
+            .Distinct();
+
+        private interface ISoftwareContent
+        {
+            string AliasName { get; }
+        }
+    }
+}
+"
+                }, // inside a property getter
+                new object[]
+                {
+                    @"
+using System.Collections.Generic;
+using System.Linq;
+
+namespace NationalInstruments.Analyzers.UnitTests
+{
+    class Test
+    {
+        void Foo(IEnumerable<ISoftwareContent> softwareContent)
+        {
+            softwareContent.Select(content => content
+                .Children.GroupBy(software => software.AliasName)
+                     .Select(g => g.First())
+                     .Distinct());
+        }
+
+        private interface ISoftwareContent
+        {
+            IEnumerable<ISoftwareContent> Children { get; }
+
+            string AliasName { get; }
+        }
+    }
+}
+"
+                } // inside an argument
+            };
+
+        public static IEnumerable<object[]> PoorlySplitStatementInVariousContextsWithDiagnosticLocation =>
+            new[]
+            {
+                new object[]
+                {
+                    @"
+using System.Collections.Generic;
+using System.Linq;
+
+namespace NationalInstruments.Analyzers.UnitTests
+{
+    class Test
+    {
+        void Foo(IEnumerable<ISoftwareContent> selectedSoftwareContents)
+        {
+            var distinctSelectedSoftware = selectedSoftwareContents.GroupBy(software => software.AliasName).Select(g => g.First()).Distinct();
```

Comment:
What about a test case where you have multiple dotted methods but the lambdas aren't next to each other:

```
GroupBy(software => software.AliasName).Distinct().Select(g => g.First()).Distinct()
```

I would assume the rule should catch this as well?

Reply:
Yes. Test [here](https://github.com/ni/csharp/blob/users/tsharma/add-long-dotted-chain-analyzer/tests/NationalInstruments.Analyzers.UnitTests/ChainOfMethodsWithLambdasAnalyzerTests.cs#L331).